### ISSUE TITLE
[feat] Automatically sync milestone/develop branches with main

### ISSUE THREAD

**Issue by brandonlenz** [trussworks/react-uswds#1011]
> **NOTE:** Depends on https://github.com/trussworks/react-uswds/pull/1026

**Is your feature request related to a problem? Please describe.**

Currently we're exploring documenting and using a flexible branching strategy where we are able to parallelize work that would not make sense (or would cause issues for consumers) to release out of order. See additional context for details on the branching strategy.

This involves keeping potentially multiple development branches up to date, which introduces easily automate-able overhead.

**Describe the solution you'd like**

Using automation, pushes to main should trigger merges to our defined develop branches.

We can very easily identify/track branches and their sync dependencies in a format similar to the following:

```
branch_sync_tree:
  - main:
    - V2
    - uswds-2.9.0:
      - uswds-2.10.0
```

From the above we should expect that pushes to `main` should trigger a total of 3 merges. 

Changes to main immediately trigger two automatic merge jobs:

1. merge `main` into `V2`. 
2. merge `main` into `uswds-2.9.0`

Additionally, since `uswds-2.9.0` is the base branch for `uswds-2.10.0`, a third merge job is triggered once `main` is merged into `uswds-2.9.0`:

3. merge `uswds-2.9.0` into `uswds-2.10.0`

This way changes to main are propagated to each additional develop branch, keeping them up-to-date.

**Describe alternatives you've considered**

Github actions and CircleCI seem to have the capability to let us do this. There may be some trickiness with auth tokens, particularly for pushing to protected branches (which ideally our develop branches should be), so we may need to involve a the truss infra bot to get around that. It may be nice to involve a bot (or something) automatically opening PRs for these merges, and auto-merging if no conflicts exist. Then human action required to resolve the conflicts would be 

I'm open to a wide range of solutions. I think the hardest part will be choosing the best one that aligns with our preferred technologies for this sort of thing.

**Additional context**

- See https://docs.google.com/document/d/1FGW1u3W6T1M-HSw8Jw3xcjM_8HdnOsV5rdbYphPuLxY/edit

**Comment by brandonlenz** [comment]
Closing, as we're not pursuing this development model currently
